workflows/pkg/convert/convert.go

```diff
@@ -98,3 +103,139 @@ func ToPipelineRun(w *v1alpha1.Workflow) (*pipelinev1beta1.PipelineRun, error) {
+	pr, err := ToPipelineRun(w)
+	if err != nil {
+		return nil, err
+	}
+
+	params := []triggersv1beta1.ParamSpec{}
+	for _, p := range w.Spec.Params {
+
+		// Triggers does not support array values from bindings
+		if p.Type == pipelinev1beta1.ParamTypeArray {
+			continue
+		}
+
+		params = append(params, triggersv1beta1.ParamSpec{
+			Name:        p.Name,
+			Description: p.Description,
+			Default:     ptr.String(p.Default.StringVal),
+		})
+		for i, prp := range pr.Spec.Params {
+			if prp.Name == p.Name {
+				pr.Spec.Params[i].Value.StringVal = fmt.Sprintf("$(tt.params.%s)", prp.Name)
+				pr.Spec.Params[i].Value.Type = pipelinev1beta1.ParamTypeString
+			}
+		}
+	}
+
+	prJson, err := json.Marshal(pr)
+	if err != nil {
+		return nil, err
+	}
+
+	tt := &triggersv1beta1.TriggerTemplate{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "TriggerTemplate",
+			APIVersion: triggersv1beta1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("tt-%s", w.Name),
+			Namespace: w.Namespace,
+		},
+		Spec: triggersv1beta1.TriggerTemplateSpec{
+			Params: params,
+			// Look in triggers code base for what this should look like
+			ResourceTemplates: []triggersv1beta1.TriggerResourceTemplate{{
+				RawExtension: runtime.RawExtension{
+					Raw: prJson,
+				},
+			}},
+		},
+	}
+
+	return tt, nil
+}
+
+// ToTriggers creates a new Trigger with inline bindings and template for each type
+// TODO: Reuse same triggertemplate for efficiency?
+func ToTriggers(w *v1alpha1.Workflow, workflowNamespace string) ([]triggersv1beta1.Trigger, error) {
+	tt, err := ToTriggerTemplate(w)
+	if err != nil {
+		return nil, err
+	}
+	triggers := []triggersv1beta1.Trigger{}
+	for i, t := range w.Spec.Triggers {
+		name := t.Name
+		if name == "" {
+			// FIXME: What if user re-orders triggers
+			// Name field should always exist -> Add it in defautls
+			name = fmt.Sprintf("%d", i)
+		}
+
+		secretToJson, err := ToV1JSON(t.Event.Secret)
+		if err != nil {
+			return nil, err
+		}
+		eventTypesJson, err := ToV1JSON([]string{t.Event.Type})
+		if err != nil {
+			return nil, err
+		}
+		// Add an interceptor to validate the payload from GitHub webhook
+		payloadValidation := triggersv1beta1.TriggerInterceptor{
+			Name: ptr.String("validate-webhook"),
+			Ref: triggersv1beta1.InterceptorRef{
+				Name: "github",
+				Kind: "ClusterInterceptor",
+			},
+			Params: []triggersv1beta1.InterceptorParams{{
+				Name:  "secretRef",
+				Value: secretToJson,
+			}, {
+				Name:  "eventTypes",
+				Value: eventTypesJson,
+			}},
+		}
+		interceptors := []*triggersv1beta1.TriggerInterceptor{&payloadValidation}
+		interceptors = append(interceptors, t.Interceptors...)
+
+		triggers = append(triggers, triggersv1beta1.Trigger{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Trigger",
+				APIVersion: triggersv1beta1.SchemeGroupVersion.String(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("%s-%s", w.Name, name),
```

Comment:
probably would also make sense to add an owner ref

Reply:
Good idea!